packages/mdc-textfield/foundation.js

```diff
@@ -65,6 +75,10 @@ export default class MDCTextfieldFoundation extends MDCFoundation {
     this.adapter_.registerInputBlurHandler(this.inputBlurHandler_);
     this.adapter_.registerInputInputHandler(this.inputInputHandler_);
     this.adapter_.registerInputKeydownHandler(this.inputKeydownHandler_);
+    this.adapter_.registerTransitionEndHandler(this.transitionEndHandler_);
+    ['mousedown', 'touchstart'].forEach((evtType) => {
```

Comment:
We do not want to register on keydown or focus. Those will cause the bottom line to emanate from the center of the textfield. It is only when `focus` is preceded by a `mousedown` or `touchstart` that we want to calculate the origin for the bottom line.

What is the advantage you see of using `focus trap` for this?